### PR TITLE
feat: Add anonymous usage tracking.

### PR DESCRIPTION
**Pull request by icycodes**
Complete TAB-69

### CODE REVIEW

File: clients/tabby-agent/src/TabbyAgent.ts [hunk 1]
```diff
@@ -140,6 +141,12 @@ export class TabbyAgent extends EventEmitter implements Agent {
     if (options.config) {
       await this.updateConfig(options.config);
     }
+    if (!this.config.anonymousUsageTracking.disable) {
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/TabbyAgent.ts:144]: should this flag being put inside of `anonymousUsageLogger`'s `event` function?

File: clients/tabby-agent/src/TabbyAgent.ts [hunk 2]
```diff
@@ -140,6 +141,12 @@ export class TabbyAgent extends EventEmitter implements Agent {
...
+      await anonymousUsageLogger.event({
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/TabbyAgent.ts:145]: is this the only event we're logging for now?

Comment (icycodes) [reply clients/tabby-agent/src/TabbyAgent.ts:145]: Yes

File: clients/tabby-agent/src/TabbyAgent.ts [hunk 3]
```diff
@@ -140,6 +141,12 @@ export class TabbyAgent extends EventEmitter implements Agent {
...
+        event: "Initialize Agent",
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/TabbyAgent.ts:146]: i believe the signature matches https://posthog.com/docs/libraries/js#send-custom-events-with-posthogcapture

File: clients/tabby-agent/src/TabbyAgent.ts [hunk 4]
```diff
@@ -141,12 +144,9 @@ export class TabbyAgent extends EventEmitter implements Agent {
     if (options.config) {
       await this.updateConfig(options.config);
     }
-    if (!this.config.anonymousUsageTracking.disable) {
-      await anonymousUsageLogger.event({
-        event: "Initialize Agent",
-        client: options.client,
-      });
-    }
+    await this.anonymousUsageLogger.event("Initialize Agent", {
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/TabbyAgent.ts:147]: Let's name the event in PascalCase: `AgentInitialized`

**Comment by wsxiaoys** [comment]
Otherwise LGTM

File: clients/tabby-agent/src/TabbyAgent.ts [hunk 5]
```diff
@@ -144,7 +144,7 @@ export class TabbyAgent extends EventEmitter implements Agent {
     if (options.config) {
       await this.updateConfig(options.config);
     }
-    await this.anonymousUsageLogger.event("Initialize Agent", {
+    await this.anonymousUsageLogger.event("Agent:Initialize", {
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/TabbyAgent.ts]: Pascal case means `AgentInitialized`, there shoud not be `:`

Comment (icycodes) [reply clients/tabby-agent/src/TabbyAgent.ts]: Fixed

**Pull request merged**